components/renku_data_services/secrets/low_level_db.py

```diff
@@ -0,0 +1,86 @@
+"""Database repo for secrets."""
+
+from collections.abc import AsyncGenerator, Callable, Sequence
+from datetime import UTC, datetime
+from typing import cast
+
+from sqlalchemy import select
+from sqlalchemy.ext.asyncio import AsyncSession
+from ulid import ULID
+
+from renku_data_services.base_api.auth import APIUser, only_authenticated
+from renku_data_services.base_models.core import InternalServiceAdmin, ServiceAdminId
+from renku_data_services.errors import errors
+from renku_data_services.secrets.models import Secret
+from renku_data_services.secrets.orm import SecretORM
+
+
+class LowLevelUserSecretsRepo:
```

Comment:
Oh, I will add a comment on the class. This is because the is a import cycle otherwise between `components/renku_data_services/secrets/db.py` and `components/renku_data_services/secrets/core.py`.

I can try to make it work by having a TYPE_CHECKING import.

Reply:
Done in fd3c92c3eb834c57dccdff3f5d294ce3b1f2a6d4.